app/code/Magento/Eav/Model/Entity/Attribute/Source/AbstractSource.php

```diff
@@ -81,21 +81,37 @@ public function getOptionText($value)
     }
 
     /**
-     * Get option id.
+     * Get option id by label.
      *
      * @param string $value
      * @return null|string
      */
     public function getOptionId($value)
     {
         foreach ($this->getAllOptions() as $option) {
-            if ($this->mbStrcasecmp($option['label'], $value) == 0 || $option['value'] == $value) {
+            if ($this->mbStrcasecmp($option['label'], $value) == 0) {
                 return $option['value'];
             }
         }
         return null;
     }
-
+        
+    /**
+     * Get option id by value.
+     *
+     * @param string $value
+     * @return null|string
+     */
+    public function getOptionIdByValue($value)
```

Comment:
@arushibansal013 do we realy need this method? As it takes $value and returns $value or null?